src/LowLevelMotionController.cpp

```diff
@@ -136,6 +142,16 @@ int main(int argc, char **argv)
 
     while (ros::ok())
     {
+        // Check the safety client before updating anything
+        ROS_ASSERT_MSG(!safety_client.isFatalActive(), "low_level_motion: fatal event from safety");
+
+        if(safety_client.isSafetyActive())
+        {
+            // Override whatever the Acceleration Planner wants to do and attempt to land
+            geometry_msgs::Twist twist;
+            twist.linear.z = -0.2; // Try to descend
+            quadController.setTargetVelocity(twist);
```

Comment:
No because the only time the quad controller is updated is after this call. Thus this will always override the velocity set by the acceleration planner.

I don't think the QuadVelocity controller should be safety aware (single responsibility) and I don't want it to have a way to stop accepting velocities in case the concept of safety is ever expanded to include recovery.